src/main/java/io/confluent/connect/hdfs/DataWriter.java

```diff
@@ -451,11 +451,20 @@ public void close() {
     // data may have continued to be processed and we'd have to restart from the recovery stage,
     // make sure we apply the WAL, and only reuse the temp file if the starting offset is still
     // valid. For now, we prefer the simpler solution that may result in a bit of wasted effort.
-    for (TopicPartition tp : assignment) {
+    for (Map.Entry<TopicPartition, TopicPartitionWriter> ent : topicPartitionWriters.entrySet()) {
```

Comment:
nit: should we `topicPartitionWriters.values()` here instead?

Reply:
Yeah, that makes sense. Had to add a getter in `TopicPartitionWriter` to get the (final) `TopicPartition`, but that's easy and worth the simplicity here.